### ISSUE TITLE
node e2e - Update CoreOS image

### ISSUE THREAD

**Issue by euank** [kubernetes/kubernetes#27383]
#27208 will require Docker to be version 1.9+.

The current image, `e2e-node-coreos-stable20160531-image` is based on CoreOS 835.13.0 and thus has docker v1.8.3.

An update to the current stable or alpha CoreOS image should unblock #27208.

The process used to be roughly documented [here](https://github.com/kubernetes/kubernetes/blob/e94e1c6e3de9dc0515827ff0c38abd032baf0b90%5E/docs/devel/e2e-node-tests.md#option-1-configuring-a-new-remote-host-from-scratch-for-testing) (and [further down](https://github.com/kubernetes/kubernetes/blob/e94e1c6e3de9dc0515827ff0c38abd032baf0b90%5E/docs/devel/e2e-node-tests.md#running-tests-against-a-gce-image) on the same page).
Validate with `export IMAGES=myimage; export IMAGE_PROJECT=my-project; make test_e2e_node REMOTE=true`.

I'll get to this when I have spare cycles, but if someone wants to do it sooner that would be awesome too!

cc @yifan-gu @aaronlevy @tmrts if any of you feel like tackling it.
cc @pwittrock @vishh as well

**Comment by yujuhong** [comment]
Marking p1 since this blocks a several PRs (#27208, #27291).

**Comment by pwittrock** [comment]
@euank If you can supply a replacement image in a GCE project it would be simple for us to pull over

**Comment by euank** [comment]
For lack of anyone else doing this, I'm doing it now. I should have an image soon.

**Comment by euank** [comment]
The updated image shows a failure in one test.

```
• Failure [308.103 seconds]
[k8s.io] Kubelet
/home/esk/dev/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/framework/framework.go:660
  metrics api
  /home/esk/dev/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e_node/kubelet_test.go:161
    when querying /stats/summary
    /home/esk/dev/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e_node/kubelet_test.go:160
      it should report resource usage through the stats api [It]
      /home/esk/dev/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e_node/kubelet_test.go:159

      Timed out after 300.000s.
      Expected
          <*errors.errorString | 0xc820359670>: {
              s: "expected only one container",
          }
      to be nil

      /home/esk/dev/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e_node/kubelet_test.go:158
```

I suspect this is due to docker using `--exec-opt native.cgroupdriver=systemd` combined with scopes. My suspicion comes from the below:

```
I0615 22:23:48.890807   15344 manager.go:864] Added container: "/system.slice/docker-de598bde9963d745094d72b05bf4676665e8cde85000184540d47243b5f60b1f.scope" (aliases: [k8s_POD.d8dbe16c_stats-busybox-d158490a-3347-11e6-a5e0-42010af0000f0_e2e-tests-kubelet-test-zrs9k_d1590272-3347-11e6-b578-42010af0000f_f2de0d21 de598bde9963d745094d72b05bf4676665e8cde85000184540d47243b5f60b1f], namespace: "docker")
I0615 22:23:48.929416   15344 manager.go:864] Added container: "/init.scope/system.slice/docker-de598bde9963d745094d72b05bf4676665e8cde85000184540d47243b5f60b1f.scope" (aliases: [k8s_POD.d8dbe16c_stats-busybox-d158490a-3347-11e6-a5e0-42010af0000f0_e2e-tests-kubelet-test-zrs9k_d1590272-3347-11e6-b578-42010af0000f_f2de0d21 de598bde9963d745094d72b05bf4676665e8cde85000184540d47243b5f60b1f], namespace: "docker")
```

It might be reasonable to create an image with that setting overridden (which does cause that test to pass) and file a separate issue to fix the root cause.

@derekwaynecarr, this seems related to your systemd/kubelet work.

**Comment by yujuhong** [comment]
@derekwaynecarr @vishh, any suggestions?

**Comment by vishh** [comment]
> It might be reasonable to create an image with that setting overridden (which does cause that test to pass) and file a separate issue to fix the root cause.

+1. If you can create an image and give me read access, I can copy that image over to the testing project that the CI uses.

**Comment by goltermann** [comment]
Adding v1.3 milestone, this appears to be blocking a v1.3 issue.  Please remove if incorrect.

**Comment by derekwaynecarr** [comment]
I am not understanding the issue.  On RHEL/Fedora/CentOS, we don't have issues with using the systemd driver, but we could be carrying patches that fix issues not fixed in native docker.  I don't get why you have `system.slice` as a child of `init.scope` - that looks like a strange hierarchy I have never seen.  I would expect everything to be `system.slice/docker-zyx.scope` by default with the systemd driver.

**Comment by erictune** [comment]
@vishh can you suggest next steps on this issue?

**Comment by vishh** [comment]
@euank Do you have an updated CoreOS image that passes all node e2e?

**Comment by euank** [comment]
I've unfortunately not been able to look at this for the last couple days. Anyone should be able to build an image and validate it per above, validation just takes time.

For some context: https://github.com/kubernetes/kubernetes/blob/master/test/e2e_node/environment/setup_host.sh is what's run on the host prior to spinning it into an image.

The issues run into that are as above related to cgroup detection duplicating container data in the kubelet api. A specific next step to try is modifying the cgroupdriver option in the docker service unit (and if that works filing a further issue to investigate a better solution).

Really though, it seems likely there's a legitimate issue in docker resulting in this messy cgroup hierarchy that needs to be addressed and if there's not a trivial workaround I guess we have to decide if that's a release blocker or not.

**Comment by yifan-gu** [comment]
I will be looking into this one shortly.

**Comment by derekwaynecarr** [comment]
I am not sure what version of docker is running and what cgroup driver it is configured to utilize. but 

I did a quick pass, and I think we carry the following cgroup related patches in docker 1.10.x for RHEL family:

https://github.com/opencontainers/runc/pull/648
https://github.com/opencontainers/runc/pull/491
https://github.com/opencontainers/runc/pull/511

It is still a mystery to me why or how you have system.slice as a child of init.scope in that one message.

For clarity, the systemd/kubelet work I did is unrelated.  It just verified that the container runtime was running in a unified cgroup, and the messages here are related to the pods not the container runtime.

**Comment by aaronlevy** [comment]
Just as a side note, that the addition of `--exec-opt native.cgroupdriver=systemd` was prompted by an issue with the kubelet originally (CoreOS bumped to v1.10 prior to k8s technically supporting it).

https://github.com/coreos/bugs/issues/1132
https://github.com/coreos/coreos-kubernetes/issues/293
https://github.com/coreos/coreos-overlay/pull/1797

I'm not sure the current state of the v1.10 patch set, or if we should carry additional patches, or if we should now revert this change in the OS (if we're going to be overriding it to get tests to pass anyway).

/cc @crawford @philips

**Comment by yifan-gu** [comment]
It seems that `CoreOS stable 1010.5.0` could be good for this. It has systemd 225(which is below 226, where init.scope is not introduced yet), and docker 1.10.3
The cgroup also looks sane:
```
$ sudo  docker run -it --rm busybox /bin/sh
/ # cat /proc/self/cgroup
9:memory:/system.slice/docker-e86af217c78a4c3f79b7b7db7df1e01fbe3aeec2fe21da0a4934c6de8736000e.scope
8:blkio:/system.slice/docker-e86af217c78a4c3f79b7b7db7df1e01fbe3aeec2fe21da0a4934c6de8736000e.scope
7:freezer:/system.slice/docker-e86af217c78a4c3f79b7b7db7df1e01fbe3aeec2fe21da0a4934c6de8736000e.scope
6:devices:/system.slice/docker-e86af217c78a4c3f79b7b7db7df1e01fbe3aeec2fe21da0a4934c6de8736000e.scope
5:cpuset:/system.slice/docker-e86af217c78a4c3f79b7b7db7df1e01fbe3aeec2fe21da0a4934c6de8736000e.scope
4:perf_event:/system.slice/docker-e86af217c78a4c3f79b7b7db7df1e01fbe3aeec2fe21da0a4934c6de8736000e.scope
3:cpu,cpuacct:/system.slice/docker-e86af217c78a4c3f79b7b7db7df1e01fbe3aeec2fe21da0a4934c6de8736000e.scope
2:net_cls,net_prio:/system.slice/docker-e86af217c78a4c3f79b7b7db7df1e01fbe3aeec2fe21da0a4934c6de8736000e.scope
1:name=systemd:/system.slice/docker-e86af217c78a4c3f79b7b7db7df1e01fbe3aeec2fe21da0a4934c6de8736000e.scope
```

Try running node e2e on this.

**Comment by yifan-gu** [comment]
```
$ test/e2e_node/jenkins/e2e-node-jenkins.sh test/e2e_node/jenkins/template.properties
+ : test/e2e_node/jenkins/template.properties
+ . test/e2e_node/jenkins/template.properties
++ GCE_HOSTS=instance-2
++ GCE_IMAGES=
++ GCE_ZONE=us-east1-b
++ GCE_PROJECT=gce-coreos-testing
++ GCE_IMAGE_PROJECT=
++ CLEANUP=true
++ SETUP_NODE=false
+ go build test/e2e_node/environment/conformance.go
+ WORKSPACE=/tmp/
+ ARTIFACTS=/tmp//_artifacts
+ mkdir -p /tmp//_artifacts
+ go run test/e2e_node/runner/run_e2e.go --logtostderr '--vmodule=*=2' --ssh-env=gce --zone=us-east1-b --project=gce-coreos-testing --image-project= --hosts=instance-2 --images= --cleanup=true --results-dir=/tmp//_artifacts --ginkgo-flags= --setup-node=false
/tmp/go-build918203318/command-line-arguments/_obj/exe/run_e2e flag redefined: log_dir
panic: /tmp/go-build918203318/command-line-arguments/_obj/exe/run_e2e flag redefined: log_dir

goroutine 1 [running]:
panic(0x1784860, 0xc8203acc30)
	/usr/local/go/src/runtime/panic.go:481 +0x3e6
flag.(*FlagSet).Var(0xc820060060, 0x7f088e5896a8, 0xc8203acb60, 0x1ec9a58, 0x7, 0x20b1c60, 0x2f)
	/usr/local/go/src/flag/flag.go:776 +0x454
flag.(*FlagSet).StringVar(0xc820060060, 0xc8203acb60, 0x1ec9a58, 0x7, 0x0, 0x0, 0x20b1c60, 0x2f)
	/usr/local/go/src/flag/flag.go:679 +0xc7
flag.(*FlagSet).String(0xc820060060, 0x1ec9a58, 0x7, 0x0, 0x0, 0x20b1c60, 0x2f, 0xc8203acb40)
	/usr/local/go/src/flag/flag.go:692 +0x83
flag.String(0x1ec9a58, 0x7, 0x0, 0x0, 0x20b1c60, 0x2f, 0x28)
	/usr/local/go/src/flag/flag.go:699 +0x5f
github.com/golang/glog.init()
	/home/yifan/gopher/src/github.com/golang/glog/glog_file.go:41 +0x13e
main.init()
	/home/yifan/kubernetes_kubeup_master/test/e2e_node/runner/run_e2e.go:356 +0x74
```
Ideas ? @vishh @pwittrock

**Comment by yifan-gu** [comment]
```
Ran 23 of 23 Specs in 199.033 seconds
SUCCESS! -- 23 Passed | 0 Failed | 0 Pending | 0 Skipped PASS
```

**Comment by yifan-gu** [comment]
BTW, the host port doesn't work for both runtime with the cni plugin

**Comment by yujuhong** [comment]
#27913 has been merged. Closing.
